core/templates/dev/head/domain/topic/TopicValidationService.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to validate the consistency of a topic. These
+ * checks are performable in the frontend to avoid sending a potentially invalid
+ * topic to the backend, which performs similar validation checks to these
+ * in topic_domain.Topic and subsequent domain objects.
+ */
+
+oppia.factory('TopicValidationService', [
+  function() {
+    var _validateSubtopic = function(subtopic) {
```

Comment:
You know -- I just realized that these really ought to be subtopic.validate() and topic.validate(), since that way they're more likely to get updated when the definition of the subtopic/topic changes. But let's get this in and fix it later -- can you file an issue for it?

Reply:
Yeah, will file an issue.

Reply:
Done, created [issue](https://github.com/oppia/oppia/issues/5202).